src/rgw/rgw_realm.cc

```diff
@@ -228,7 +73,7 @@ int RGWRealm::find_zone(const DoutPrefixProvider *dpp,
   epoch_t epoch = 0;
 
   RGWPeriod period(period_id, epoch);
-  int r = period.init(dpp, cct, sysobj_svc, get_id(), y);
+  int r = cfgstore->read_period(dpp, y, period_id, epoch, period);
```

Comment:
> WARNING: period init failed

i think this is because we're trying to read an empty `period_id`. before, we were passing the realm id (`RGWRealm::get_id()`) to `RGWPeriod::init()`, where it was loading that realm to learn its `current_period`

i guess this is coming from `RGWSI_Zone::search_realm_with_zone()`, which has already loaded this realm. so we should be able to use the member variable `current_period` directly:
```c++
  RGWPeriod period;
  int r = cfgstore->read_period(dpp, y, current_period, std::nullopt, period);
```